cl/search/api_serializers.py

```diff
@@ -310,17 +311,71 @@ def create(self, validated_data):
         """
-        fields = {
-            "snippet": serializers.CharField(read_only=True),
-            "panel_ids": serializers.ListField(read_only=True),
+        mapping_fields = {
+            AudioDocument: {
+                "snippet": serializers.CharField(read_only=True),
+                "panel_ids": serializers.ListField(read_only=True),
+            },
+            PersonDocument: {
+                "snippet": serializers.CharField(read_only=True),
+                "aba_rating": serializers.ListField(read_only=True),
+                "alias_ids": serializers.ListField(read_only=True),
+                "appointer": serializers.ListField(read_only=True),
+                "court": serializers.ListField(read_only=True),
+                "court_exact": serializers.ListField(read_only=True),
+                "races": serializers.ListField(read_only=True),
+                "school": serializers.ListField(read_only=True),
+                "political_affiliation": serializers.ListField(read_only=True),
+                "political_affiliation_id": serializers.ListField(
+                    read_only=True
+                ),
+                "position_type": serializers.ListField(read_only=True),
+                "supervisor": serializers.ListField(read_only=True),
+                "predecessor": serializers.ListField(read_only=True),
+                "date_nominated": serializers.ListField(read_only=True),
+                "date_elected": serializers.ListField(read_only=True),
+                "date_recess_appointment": serializers.ListField(
+                    read_only=True
+                ),
+                "date_referred_to_judicial_committee": serializers.ListField(
+                    read_only=True
+                ),
+                "date_judicial_committee_action": serializers.ListField(
+                    read_only=True
+                ),
+                "date_hearing": serializers.ListField(read_only=True),
+                "date_confirmation": serializers.ListField(read_only=True),
+                "date_start": serializers.ListField(read_only=True),
+                "date_granularity_start": serializers.ListField(
+                    read_only=True
+                ),
+                "date_retirement": serializers.ListField(read_only=True),
+                "date_termination": serializers.ListField(read_only=True),
+                "date_granularity_termination": serializers.ListField(
+                    read_only=True
+                ),
+                "judicial_committee_action": serializers.ListField(
+                    read_only=True
+                ),
+                "nomination_process": serializers.ListField(read_only=True),
+                "selection_method": serializers.ListField(read_only=True),
+                "selection_method_id": serializers.ListField(read_only=True),
+                "termination_reason": serializers.ListField(read_only=True),
+            },
         }
```

Comment:
If there's a way to avoid this too, that'd be great, but I don't know how easy it is.

Reply:
At first, I thought we were using this mapping because we didn't have a helper to identify the document fields and select a serializer class. However, this is not the case, we're using this mapping because those fields are not defined in the `ES Document`(We add all of those fields to the `PersonDocument` query results by using the `merge_unavailable_fields_on_parent_document` method).

I refactored this class to remove the mapping and use the DRF approach to [specify fields explicitly](https://www.django-rest-framework.org/api-guide/serializers/#specifying-fields-explicitly), so we can define serializer classes for documents like these:


```python
class PersonESResultSerializer(DocumentSerializer):
    """The serializer for Person results."""

    nomination_process = serializers.ListField(read_only=True)

    class Meta:
        document = PersonDocument
        exclude = ("text", "person_child")
```

In the example above, the class will serializer all the fields defined in the `PersonDocument` except for "text" and "person_child" and will also include "nomination_process" when deserializing objects.